WcaOnRails/app/views/competitions/_index_form.html.erb

```diff
@@ -35,21 +35,29 @@
 </div>
 
 <div class="form-group">
-  <%= label_tag(:state) %>
+  <%= label_tag(t('.state')) %>
   <div id="state" class="btn-group" data-toggle="buttons">
     <label id="present" class="btn btn-primary">
       <%= radio_button_tag "state", "present" %>
-      <span class="caption">Present</span>
+      <span class="caption"><%= t '.present' %></span>
     </label>
     <ul class="dropdown-menu years">
       <%= hidden_field_tag "year", params[:year] %>
       <% @years.each do |year| %>
-        <li class="year"><%= link_to year.to_s.capitalize, "#", data: { year: year } %></li>
+        <% year_label = year.is_a?(Integer) ? year.to_s : t('.all_years') %>
+        <li class="year"><%= link_to year_label, "#", data: { year: year } %></li>
       <% end %>
     </ul>
     <label id="past" class="btn btn-primary" data-toggle="dropdown">
       <%= radio_button_tag "state", "past" %>
-      <span class="caption">Past <%= "from #{params[:year]}" if @past_selected %></span>
+      <span class="caption">
+        <% if @past_selected %>
+          <% #Here params[:year] is a string, so we need regexp %>
+          <%= params[:year] =~ /\A\d+\z/ ? t('.past_from', year: params[:year]) : t('.past_all') %>
```

Comment:
So here I had to differentiate "Past competitions" from "Past competitions from XXXX" from "Past competitions from all years", because it was just a pain to make French match the initial flow of the sentence ("Past competitions" + something).
I think the issue may appear in other languages as well, so splitting the strings may just be the right solution.